### PR TITLE
passky-desktop: 7.1.0 -> 8.1.1

### PR DESCRIPTION
**Pull request by akkesm**
###### Description of changes

I managed to package this version from source, no more extracting appimage/dmg files.
Thoroughly tested.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by akkesm** [comment]
Result of `nixpkgs-review pr 207730` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>passky-desktop</li>
  </ul>
</details>

**Comment by akkesm** [comment]
Changed to the latest release 8.0.1 -> 8.1.1

**Comment by akkesm** [comment]
Result of `nixpkgs-review pr 207730` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>passky-desktop</li>
  </ul>
</details>

**Review by pbsds — COMMENTED** [review]
Good work!

This version make the cursor on my 200% scaling (hidpi) machine look small / unscaled. The prior version did not have this issue, and I run other electron apps which does not experience this issue. Could you look into this?

**Comment by akkesm** [comment]
I'm sorry, I can't reproduce the scaling issue. The cursor looks exactly the same as before in my tests. Granted, I don't have a hidpi display, so I'm using a VM with increased resolution and scaling on a 1080p host display.
Do you still have this issue?

File: pkgs/applications/misc/passky-desktop/default.nix [hunk 1]
```diff
@@ -1,59 +1,66 @@
...
-  meta = with lib; {
-    homepage = "https://passky.org";
-    downloadPage = "https://github.com/Rabbit-Company/Passky-Desktop/releases";
-    license = licenses.gpl3Only;
-    maintainers = with maintainers; [ akkesm ];
-    platforms = builtins.attrNames srcs;
+  src = fetchFromGitHub {
+    owner = "Rabbit-Company";
+    repo = "Passky-Desktop";
+    rev = "v${version}";
```
[the remark this answers is not in the document]
Comment (akkesm) [reply pkgs/applications/misc/passky-desktop/default.nix]: Done, thanks.

**Comment by akkesm** [comment]
I have contacted the author of the app regarding #258048, though this PR should aready fix it since it uses electron from nixpkgs. Please correct me if I'm wrong.

If packaging from source isn't viable, I will edit the PR as soon as an update is released.

File: pkgs/applications/misc/passky-desktop/default.nix [hunk 2]
```diff
@@ -1,59 +1,66 @@
...
+  desktopItem = makeDesktopItem {
+    name = "Passky";
+    type = "Application";
+    desktopName = "passky";
+    comment = "Simple, modern, open source and secure password manager.";
+    icon = "passky";
+    exec = "passky %U";
+    categories = [ "Utility" ];
+    startupWMClass = "Passky";
+  };
```
Comment (pbsds) [inline pkgs/applications/misc/passky-desktop/default.nix:54]:
```suggestion
  desktopItems = [(makeDesktopItem {
    name = "Passky";
    type = "Application";
    desktopName = "passky";
    comment = "Simple, modern, open source and secure password manager.";
    icon = "passky";
    exec = "passky %U";
    categories = [ "Utility" ];
    startupWMClass = "Passky";
  })];
```

File: pkgs/applications/misc/passky-desktop/default.nix [hunk 3]
```diff
@@ -1,59 +1,66 @@
...
+      resolution=''${icon%".png"}
+      resolution=''${resolution##*/icon-}
+      mkdir -p "$out/share/icons/hicolor/''${resolution}/apps"
+      ln -s "$icon" "$out/share/icons/hicolor/''${resolution}/apps/passky.png"
+    done
 
-  darwin = stdenv.mkDerivation {
-    inherit pname version src meta;
+    mkdir "$out/share/applications"
+    ln -s ${desktopItem}/share/applications/* "$out/share/applications"
```
Comment (pbsds) [inline pkgs/applications/misc/passky-desktop/default.nix:37]:
```suggestion
```

Comment (pbsds) [reply pkgs/applications/misc/passky-desktop/default.nix:37]: The problem with this approach is that it is not overridable

File: pkgs/applications/misc/passky-desktop/default.nix [hunk 4]
```diff
@@ -1,59 +1,66 @@
...
+    rev = "refs/tags/v${version}";
+    sha256 = "1ma8s1bngjdzclcc4m5pcmavk38sidaskmz7dgfnv84y35si18dr";
   };
 
-  linux = appimageTools.wrapType2 {
-    inherit pname version src meta;
-
-    extraInstallCommands = ''
-      mv $out/bin/${pname}-${version} $out/bin/${pname}
+  nativeBuildInputs = [ makeWrapper ];
```
Comment (pbsds) [inline pkgs/applications/misc/passky-desktop/default.nix:20]:
```suggestion
  nativeBuildInputs = [ makeWrapper, copyDesktopItems ];
```

**Review by pbsds — CHANGES_REQUESTED** [review]
This is looking very good. Some minor nits

File: pkgs/applications/misc/passky-desktop/default.nix (hunk 3 above)
Comment (akkesm) [reply pkgs/applications/misc/passky-desktop/default.nix]: Fixed it. Honestly, I didn't know copyDesktopItems was a thing, I would have used it from the start. Thanks for pointing this out.

**Comment by pbsds** [comment]
LGTM

---

Result of `nixpkgs-review pr 207730` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>passky-desktop</li>
  </ul>
</details>

---

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

**Review by pbsds — APPROVED** [review]

**Pull request merged**